Feature/1176142788936875/trash bin

Alfredo-Moreira opened the pull request:
## Description

<!-- Describe your changes here -->

## Asana ticket:

## PR Checklist

<!-- Please validate your changes with the checklist below before marking for code review. -->

- [ ] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [ ] If your PR is not a hotfix, is it targeted for `dev`? If your PR is a hotfix, is it targeted for `master`?
- [ ] Unit and functional test coverage was added where applicable.
- [ ] CI/CD passes for your PR.
- [ ] Complex code is well documented with comments.
- [ ] Does the original ticket have test instructions? If not add them below

## How to Test

<!-- Provide instructions for how to test/validate the changes. -->

FJKhan commented on cypress/integration/dashboard_spec.js:
```diff
@@ -7,7 +7,7 @@
 describe('Home Page Dashboard Tests', () => {
 
-    let viewports = [Cypress.env('desktop'),Cypress.env('tablet'),Cypress.env('mobile')];
+    let viewports = [Cypress.env('desktop')];
```
are these changes deliberate?

Alfredo-Moreira replied:
Yes, So far from what i have seen there are no code changes between Desktop, tablet and mobile. The app just resizes to fit the screen. It was running the same test 3 times. I can however bring them back.

FJKhan commented on cypress/support/reusable_tests/admin.js:
```diff
@@ -87,31 +88,40 @@ Cypress.Commands.add('testAdminPageElements',(viewport,creds)=>{
     cy.viewport(viewport);
     cy.login(creds.email,creds.password);
-
+    var type = 'lawyer';
```
any particular reason we're using the lawyer user type here?

Alfredo-Moreira replied:
Just one of the options available that is not a regular user.

Alfredo-Moreira replied:
Do you want me to test with a random selection of the options? I can do that and not test with a static selection.

Alfredo-Moreira replied:
I added all possible types and tested it

FJKhan commented on cypress/support/reusable_tests/admin.js:
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
+    //Click
+    cy.getElementByTestId('header-admin-link').click();
+     
+    cy.wait(['@deletedOrganizations','@deletedOrgServices'])
```
I think this wait is in the wrong place. it should go after said actions are invoked e.g. whatever action triggers the network request

Alfredo-Moreira replied:
The cy.login triggers a response. Immediately after the login, the redirect to dashboard triggers the organizations endpoint.

Alfredo-Moreira replied:
Admin Link- routes to endpoint( /admin ), this endpoint does all the calls at once for all 3 available tabs(Users,Suggestions,Trash Bin) under that option. So I have to intercept after that click

FJKhan commented on cypress/support/reusable_tests/admin.js:
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
+
+    cy.get('@deletedOrgServices').then(interception =>{
+        if(interception.response.body.organizations.length){
+            if(interception.response.body.organizations.length){
```
duplicate if statement

Alfredo-Moreira replied:
nice catch!

Alfredo-Moreira replied:
Removed

FJKhan commented:
1. can we change "View Deleted Service Organization" to just "View organization" ? it's lengthy & confusing

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/7406914/142043684-cb8647a3-8e3a-4a36-ade3-649ebc9fb3d7.png">

2. catalog is still showing deleted service. we should update the api to filter out the deleted orgs/services as well
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/7406914/142044099-6b3f86dc-ba16-499f-8c0c-ea374b5f4a73.png">

3. I think we should include the service name in the restore service dialog 
<img width="509" alt="image" src="https://user-images.githubusercontent.com/7406914/142044334-365f3eb6-1a4f-4480-8a6f-adb2636e4f65.png">

Alfredo-Moreira commented on cypress/support/reusable_tests/admin.js:
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
+    cy.getElementByTestId('admin-tab-trash-bin').click();
+
+    cy.get('@deletedOrganizations').then(interception =>{
+        if(!interception.response.body.organizations.length){
```
I can update this.

Alfredo-Moreira replied:
In order to address this, I will have to create a function to clear deleted orgs/services and set them to is_deleted: false to test the empty state, and then and org that is deleted and has deleted services.

Alfredo-Moreira replied:
https://github.com/asylum-connect/control-panel/pull/69/commits/9c12ec4b1128256d673c392446de1fbc37c5533a

Alfredo-Moreira commented on src/components/AdminPanelTrashBin.js:
```diff
@@ -0,0 +1,183 @@
+    const {closeModal, openModal} = useContext(ContextFormModal);
+    const deletedOrgs = useAPIGet(`/organizations?deleted=true`)?.data?.organizations;
+    const deletedServices = useAPIGet(`/organizations?serviceDeleted=true`);
+    const reducedDeletedServices = deletedServices?.data?.organizations?.reduce(
```
Yes, by default the organizations api (/organizations) returns organizations that are not deleted, and without deleted services. By using the query param, I can force that case scenario and populate the trash bin.

Alfredo-Moreira replied:
But i will take a look at that. I think I might have had that there before the changes I implemented in the API

Alfredo-Moreira replied:
Addressed the reduce by reduce the if statement with https://github.com/asylum-connect/control-panel/pull/69/commits/cd38af148e3a772ad2fd336727b3f90d1adc7c08 . But I use the reduce to make the array with services

Alfredo-Moreira replied:
The API currently only addresses that with [v1/organizations/:id](https://github.com/asylum-connect/catalog-api/blob/dev/src/routes/organizations.js#L209) query. Unfortunately, the [v1/organizations?query](https://github.com/asylum-connect/catalog-api/blob/dev/src/routes/organizations.js#L19) type of endpoint is a straight query and not an aggregate pipeline. This is why I'm using the reduce to filter out the services that are not deleted. That query only returns all the orgs that have deleted services. Turning that into an aggregate would an huge undertaking, as the call handles a lot of other query params.

Alfredo-Moreira reviewed (commented):
Saw all your comments. I will begin addressing them.

Alfredo-Moreira commented:
> 

Addressed:
 1. Addressed it by removing extra text
 2. Looking into this
 3. Had to add message display support on the formModal. Not in body of modal it will display the name of the organization or service being deleted or restored.
